internal/impl/timeplus/output.go

```diff
@@ -0,0 +1,215 @@
+package timeplus
+
+import (
+	"context"
+	"errors"
+	"sort"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/redpanda-data/connect/v4/internal/impl/timeplus/http"
+)
+
+var outputConfigSpec *service.ConfigSpec
+
+func init() {
+	// TODO: add Version
+	outputConfigSpec = service.NewConfigSpec().
+		Categories("Services").
+		Summary("Sends message to a Timeplus Enterprise stream").
+		Description(`
+This output can send message to Timeplus Enterprise Cloud, Timeplus Enterprise Onprem or directly to timeplusd.
+
+This output accepts structured message only. It also expects all message contains the same keys and matches the schema of stream. If the upstream source returns
+unstructured message such as string, please include a processors to contruct the strcuture message. Here is a short sample:
+
+` + "```yml" + `
+output:
+  timeplus:
+    workspace: my_workspace_id
+    stream: mystream
+    apikey: JFmlmLDJPtrgo6cEgSDKHyg9NknZ5PM2M9lGelCwE2BtHPxlXnCbQaWua
+
+	processors:
+	  - mapping: |
+        root = {}
+        root.raw = content().string() # Make sure your stream contains a string type raw field
+` + "```" + `
+
+A sample config to send the data to Timeplus Enterprise Cloud
+` + "```yml" + `
+output:
+  timeplus:
+    workspace: my_workspace_id
+    stream: mystream
+    apikey: JFmlmLDJPtrgo6cEgSDKHyg9NknZ5PM2M9lGelCwE2BtHPxlXnCbQaWua-9a
+` + "```" + `
+
+A sample config to send the data to Timeplus Enterprise Onprem
+` + "```yml" + `
+output:
+  timeplus:
+    url: http://localhost:8000
+    workspace: my_workspace_id
+    stream: mystream
+    username: username
+    password: pw
+` + "```" + `
+
+A sample config to send the data to timeplusd
+` + "```yml" + `
+output:
+  timeplus:
+    url: http://localhost:3218
+    stream: mystream
+    username: username
+    password: pw
+` + "```" + `
+`)
+
+	outputConfigSpec.
+		Field(service.NewStringEnumField("target", http.TargetTimeplus, http.TargetTimeplusd).Default(http.TargetTimeplus)).
+		Field(service.NewURLField("url").Description("The url should always include schema and host. The port number should be included as well if it is not 80.").Examples("https://us.timeplus.cloud", "http://localhost:8000")).
+		Field(service.NewStringField("workspace").Optional().Description("ID of the workspace. Required if target is `timeplus`.")).
+		Field(service.NewStringField("stream").Description("The name of the stream. Make sure the schema of the stream matches the input")).
+		Field(service.NewStringField("apikey").Secret().Optional().Description("The API key. Required if you are sending message to Timeplus Enterprise Cloud")).
+		Field(service.NewStringField("username").Optional().Description("The username. Required if you are sending message to Timeplus Enterprise Onprem or timeplusd")).
```

Comment:
Replace Onprem with self-host, to align with website/docs

Reply:
fixed